### PR TITLE
Fix fireproof and lavaproof objects receiving burn integrity damage

### PR DESCRIPTION
**Pull request by Jewelry-x**
## About The Pull Request

Adds checks for if an object is fire proof or lava proof before applying integrity damage from burn type.
## Why It's Good For The Game

Makes fire proof and lava proof objects immune to burn type damage.
## Changelog
:cl:
fix: fix items taking burn damage even if fire proof or lava proof
/:cl:

### CODE REVIEW

File: code/game/atom/atom_defense.dm [hunk 1]
```diff
@@ -25,6 +25,8 @@
 		play_attack_sound(damage_amount, damage_type, damage_flag)
 	if(resistance_flags & INDESTRUCTIBLE)
 		return
+	if(damage_type == BURN && damage_flag != "laser" && resistance_flags & FIRE_PROOF || resistance_flags & LAVA_PROOF)
```
Comment (SmArtKar) [inline code/game/atom/atom_defense.dm:28]: "laser" has a LASER define

**Comment by SmArtKar** [comment]
You do realize how much this changes balance for mechs, correct? An incredible amount of weapons will be completely useless against them, and that's not talking about all the fireproof structures and machinery out there. Your WITGFTG isn't elaborating on the reasoning either, and its far from being a fix.

File: code/game/atom/atom_defense.dm (hunk 1 above)
Comment (SmArtKar) [reply code/game/atom/atom_defense.dm:28]: also you should wrap &s, else they break

**Comment by Jacquerel** [comment]
Please update your PR body to provide reasoning for these changes, this is a balance change and not a fix and does not have any reasoning provided for why we should do it.

**Pull request closed without merging**